code/modules/mob/living/carbon/human/species.dm

```diff
@@ -510,10 +510,16 @@
 		H.see_in_dark = 8
 		if(!H.druggy)		H.see_invisible = SEE_INVISIBLE_LEVEL_TWO
 	else
-		H.sight &= ~(SEE_TURFS|SEE_MOBS|SEE_OBJS)
+		if(!locate(SEE_TURFS,H.permanent_sight_flags))
```

Comment:
The Vast majority of your ```if(!locate(X,Y))``` can be written as ```if(!(X in Y))``` which is the standard for finding things in lists.